### ISSUE TITLE
Give plotOutput() the ability to fix an aspect ratio

### ISSUE THREAD

**Issue by hadley** [rstudio/shiny#2470]
So that you can use a variable width (e.g. 100%), and have the height dynamically adjust. 

Currently possible to work around on the R side, https://github.com/rstudio/shiny/issues/650, but ideally would happen in js.

**Comment by jcheng5** [comment]
The width/height arguments currently accept numeric values, or valid CSS sizes. How should we distinguish aspect ratio arguments?

Colon?

```r
# Set the height to whatever value results in a 16w x 9h aspect ratio
plotOutput("plot", height = "16:9")
# Set the width to whatever value results in a 16w x 9h aspect ratio
plotOutput("plot", width = "16:9", height = 600)
```

`X` suffix?

```r
# Set the height to width*0.5
plotOutput("plot", height = "0.5X")
# Set the width to height*0.5
plotOutput("plot", width = "0.5X")
```

**Comment by jcheng5** [comment]
And you'd get an error if you tried to specify both width and height as aspect ratios.

**Comment by cpsievert** [comment]
Another option is to take knitr's `fig.asp` (via a new argument):

> fig.asp: (NULL; numeric) the aspect ratio of the plot, i.e. the ratio of height/width; when fig.asp is specified, the height of a plot (the chunk option fig.height) is calculated from fig.width * fig.asp

**Comment by hadley** [comment]
Yeah, I was thinking you'd have `aspect.ratio` (or similar) which would default to `NULL`, or otherwise you'd set to `height / width` (and it would override the default height).

**Comment by jcheng5** [comment]
https://www.smashingmagazine.com/2019/03/aspect-ratio-unit-css/
